src/libmongoc/CMakeLists.txt

```
@@ -108,6 +108,33 @@ else ()
 endif ()
 
 
+configure_file (
+   "${SOURCE_DIR}/src/utf8proc-2.8.0/utf8proc.h"
+   "${CMAKE_BINARY_DIR}/src/utf8proc-2.8.0/utf8proc.h"
+   COPYONLY
+)
+set (UTF8PROC_INCLUDE_DIRS "")
+
+if ( (ENABLE_UTF8PROC STREQUAL "BUNDLED")
+   OR (ENABLE_UTF8PROC STREQUAL "AUTO" AND NOT UTF8PROC_FOUND) )
+   message (STATUS "Enabling utf8proc compression (bundled)")
+   add_library (utf8proc_obj OBJECT "${UTF8PROC_SOURCES}")
+   set_property (TARGET utf8proc_obj PROPERTY POSITION_INDEPENDENT_CODE TRUE)
+   # This tells the bundled utf8proc where to find its generated headers
+   target_include_directories (utf8proc_obj
+      BEFORE PUBLIC
+      "${SOURCE_DIR}/src/utf8proc-2.8.0"
+      "${CMAKE_BINARY_DIR}/src/utf8proc-2.8.0"
+   )
+   set (SOURCES ${SOURCES} $<TARGET_OBJECTS:utf8proc_obj>)
```

```suggestion
   set (SOURCES ${SOURCES} $<TARGET_OBJECTS:utf8proc_obj>)
   target_compile_definitions (utf8proc_obj PUBLIC UTF8PROC_STATIC)
```

Windows tasks are failing with [this log](https://parsley.mongodb.com/evergreen/mongo_c_driver_windows_2017_32_debug_compile_nosasl_nossl_patch_7d5beeae25e3c821b6e01e204ba6d323184c8b7d_6494b5b93627e04d43aaaff4_23_06_22_20_57_31/0/task?bookmarks=0,220&shareLine=140):

```
C:\data\mci\f6caccb9ade092c41e4af9bff293259e\mongoc\src\utf8proc-2.8.0\utf8proc.c(56): error C2491: 'utf8proc_utf8class': definition of dllimport data not allowed
```

I expect the dllimport attribute needs to be removed. dllimport is used by consumers to declare that symbols are defined in another DLL. The utf8proc source files are compiled directly in the C driver (not as a separate DLL). 

https://github.com/JuliaStrings/utf8proc/blob/1cb28a66ca79a0845e99433fd1056257456cef8b/utf8proc.h#L123-L125 suggests defining `UTF8PROC_STATIC` will remove the dllimport attribute.

If this does not resolve the Windows tasks, try spawning a Windows host in Evergreen and building. That may help to more quickly diagnose the issue.